### PR TITLE
HTML: set location.protocol to non-broken-non-functioning schemes

### PR DESCRIPTION
**Pull request by annevk**
See https://github.com/whatwg/url/issues/61 for context.

https://github.com/w3c/web-platform-tests/pull/4412 tests broken
schemes.

### CODE REVIEW

**Comment by annevk** [comment]
The -weird.html is an earlier iteration of the test (excluding data URL tests) that for some reason never completes running in Firefox. I thought I'd include it since it does show differences between the two ways of testing.

File: html/browsers/history/the-location-interface/location-protocol-setter-non-broken.html [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    'ftp',
+    'gopher',
+    'http+x'
+  ].forEach((val) => {
+    async_test((t) => {
+      // HTTP URL <iframe>
+      const frame = document.createElement("iframe")
+      frame.src = "resources/dummy.html"
+      frame.onload = t.step_func(() => {
+        try {
```
Comment (zcorpan) [inline html/browsers/history/the-location-interface/location-protocol-setter-non-broken.html]: Remove

File: html/browsers/history/the-location-interface/location-protocol-setter-non-broken-weird.html [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          assert_equals(self[index].location.protocol, location.protocol)
+          assert_equals(self[index].location.host, location.host)
+          assert_equals(self[index].location.port, location.port)
+          t.done()
+        }, 500)
+      } catch(e) {
+        assert_unreached(e)
+      }
+    })
+  })
```
Comment (zcorpan) [inline html/browsers/history/the-location-interface/location-protocol-setter-non-broken-weird.html]: Include `val` in the test name

**Review by zcorpan — CHANGES_REQUESTED** [review]

**Comment by annevk** [comment]
Thanks, pushed fixes.

**Review by zcorpan — APPROVED** [review]

**Pull request merged**